sliceanddice/provider/src/main/java/com/novoda/sliceprovider/SearchSlice.kt

```diff
@@ -0,0 +1,106 @@
+package com.novoda.sliceprovider
+
+import android.annotation.SuppressLint
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.graphics.drawable.Icon
+import android.net.Uri
+import android.text.Html
+import android.text.SpannableStringBuilder
+import android.text.style.ForegroundColorSpan
+import androidx.slice.Slice
+import androidx.slice.builders.GridBuilder
+import androidx.slice.builders.ListBuilder
+import androidx.slice.builders.ListBuilder.HeaderBuilder
+import androidx.slice.builders.ListBuilder.RowBuilder
+
+@SuppressLint("RestrictedApi") // The GridBuilder's image size definitions shouldn't be restricted APIs
+internal fun createSearchSlice(context: Context, sliceUri: Uri): Slice = ListBuilder(context, sliceUri)
```

Comment:
This content is hardcoded but I guess the idea behind slices is that providers can build UI components into clients by describing them in this react way?

Reply:
This has nothing to do with React; the only similarity is that you are defining the content structure and don't control here how they're rendered. The API is composed of (extremely annoying) builders, it's rather normal, the only "novel" thing is the amount of nesting one can have, I suppose.

Reply:
yep ok, by  "in a react way" I meant in a descriptive way instead of defining the specific UI elements 👍

Reply:
Well it's not much different than, say, a notification builder in that though :)